### DISCUSSION TITLE
Is server code executed with 304

### DISCUSSION THREAD

**Discussion by tobbbe** [vercel/next.js#39855] [Help]
Does/should server code run when the server return status code 304?
I understand that the server should not return anything (client should use the cache), but I cant find any info on whether the server will executed the code in an api endpoint for example.

**Comment by icyJoseph** [comment]
Mmm I am not sure, but the way the whole 304 works is by comparing the `content` sent `before` and the `content` generated `now`, usually by hashing the `content`, and if the hashing results in the same, aka, same content then respond with 304 to the browser.

This way the browser uses the content sent `before`, however the roundtrip does happen. It just doesn't include a payload, because the payload is already present in the browser.

**Comment by pm0u** [comment]
Just want to include here - I was running into this issue and thought I solved the 304 issue by adding a `Cache-Control: no-cache` header to the response - this changed the status code to 200. However, after a lot of digging I found out that my API route still was not being evaluated on a given request (namely, by including a timestamp in the response - it never changed). I found out that because my route was `GET` request and it didn't examine the `request` param, it was "Statically Evaluated". Because my API route had side effects (revalidating and updating a DB) I needed it to be forced dynamic

https://nextjs.org/docs/app/building-your-application/routing/router-handlers#dynamic-route-handlers

**Reply by pm0u** [reply]
I do feel like there should be some sort of inference here on the part of NextJS - like if it calls a fetch with `cache: no-store` (mine was) or it uses a `revalidate` function, it should be dynamic.

**Reply by pm0u** [reply]
The clue here is the empty circle (`○`) vs the lambda (`λ`) during build - if your API route has the empty circle, it is just static cache served for every response, the code will not be run on each request.
